server/execution/api/action.py

```diff
@@ -155,6 +158,38 @@ def get_perform_action_result():
         return "Missing or invalid request parameter detected.", 400
 
 
+@api.post("/action/perform/move/patient")
+@jwt_required()
+@required("new_location_id", int, RequiredValueSource.JSON)
+@required("patient_id", int, RequiredValueSource.JSON)
+@csrf.exempt
+def move_patient(patient_id: int, new_location_id: int):
+    execution, player = util.get_execution_and_player()
+    try:
+        patient = execution.scenario.patients[patient_id]
+        new_location = execution.scenario.locations[new_location_id]
+
+        if (player.location and patient.location
+            and player.location.id != patient.location.id):
+            return "Invalid request. Players is not allowed to move patient.", 418
+        # Update player location
+        r_value = leave_location()
+        if not isinstance(r_value, dict):
+            return r_value
+
+        # Reassign Patient Location
+        patient.location = new_location
```

Comment:
What about ...
1. ... the player who initiated the move, does he move along with the patient to the new location?
2. ... other players at the original patient location? Is there after the move a top level location named after the patient but without a patient or do we have to delete it?

Reply:
1. The player moves along with the patient, according to real life situations. The patient wont flow away withouth the player. That is implemented before and after l.181.

2. The original location is accessable via top level location (see test). However along with the API implementation only a leave location would be a valid move, because the location is only accessable through the patients presence. I might have some Ideas to prevent the loss of resources in the empty space, i guess that would not be advisable due to our timelimit. But, I will add a check for the patients location when performing an action. If the patient has moved and other players are still with the patient, they need to know.